arborenv/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+set(arborenv-sources
+    affinity.cpp
+    concurrency.cpp
+    default_gpu.cpp
+    private_gpu.cpp
+)
+
+if(ARB_WITH_GPU)
+    list(APPEND arborenv-sources gpu_uuid.cpp)
+endif()
+
+add_library(arborenv ${arborenv-sources})
+
+add_library(arborenv-public-headers INTERFACE)
+target_include_directories(arborenv-public-headers INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+
+# A public arbor dependency is required for the use of arb::mpi_error,
+# which in turn requires the arbor libary's singleton for the mpi_error_category
+# singleton.
+
+target_link_libraries(arborenv PUBLIC arbor arborenv-public-headers)
+target_link_libraries(arborenv PRIVATE arbor-config-defs arborenv-private-deps)
+
+if(ARB_WITH_GPU)
+    # So that cpp and hpp files can use cuda.h and cuda_runtime.h
+    target_include_directories(arborenv PRIVATE ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+
+    # The arborenv lib needs to use the CUDA NVML library for CUDA 9
+    if (${CUDA_VERSION_MAJOR} LESS 10)
```

Comment:
This search is being done in two places: it was also added to `CMakeLists.txt`.

Reply:
Yup, that's definitely an edit I forgot to do.
Moved the NVML stuff up to top because we need to know there what library dependencies to munge in the generated `arbor-config.cmake` script; forgot to remove from `arborenv/CMakeLists.txt`.